analysis/paper/paper.qmd

```diff
@@ -206,27 +207,159 @@ emu_words <- emu_words |>
 
 ```
 
-## Topic model
+### Topic model
 
-Similar to clustering on numeric data, topic modeling is a method of unsupervised classification of topics found in a group of documents [@silge2017]. We will use the Latent Dirichlet Allocation (LDA) algorithm for topic modeling. In LDA, each document is a mixture of topics and each topic is a mixture of words. The probability of a term being part of a topic is given by the $\beta$ (beta) statistic, while the probability of a document being part of a topic is given by the $\gamma$ (gamma) statistic.
+Similar to clustering on numeric data, topic modeling is a method of unsupervised classification of topics found in a group of documents [@silge2017]. We used the Latent Dirichlet Allocation (LDA) algorithm for topic modeling. In LDA, each document is a mixture of topics and each topic is a mixture of words. The probability of a term being part of a topic is given by the $\beta$ (beta) statistic, while the probability of a document being part of a topic is given by the $\theta$ (theta) statistic.
+
+The number of topics was qualitatively determined
 
 ```{r topic-model-lda}
+#| dependson: short-stop-words
 
 # Prepare document-term matrix as input for LDA model
-emu_dtm <- convert_to_dtm(emu_words, "title", "word")
 
-# Create LDA model
-emu_lda <- LDA(emu_dtm, k = 2, control = list(seed = 2023))
+# Reduce the size of the DTM by excluding words with a frequency lower than 5
+emu_dtm <- convert_to_dtm(emu_theses_words, 
+                          title_col = "title", 
+                          word_col = "word", 
+                          min_term_freq = 5)
+
+# Fit LDA model with K clusters
+# # Determine K value empirically
+# K = get_number_of_topics(emu_dtm)
+# # This returns K = 23
+# TODO run the model with K = 23 and compare with results using qualitative value
+
+# Determine K value qualitatively
+# K = 5 corresponds to the 2 TUD semesters and the 3 exchange semesters combined 
+emu_lda <- LDA(emu_dtm, k = 5, method="Gibbs",
+               control = list(seed = 2023, iter = 500, verbose = 25))
+
+# Extract beta and theta statistics from LDA model
+beta <- posterior(emu_lda)$terms
+theta <- posterior(emu_lda)$topics 
+
+# Preview 10 terms from each topic
+terms(emu_lda, 10)
+
+# Add pseudo-names to topics based on the top 5 words in each topic
+topic_names <- name_topics(beta)
+
+```
+
+
+```{r topic-model-lda}
+
+# What is the dominant topic in each document?
+N = 20
+topic_prop_examples <- theta[1:N,]
+colnames(topic_prop_examples) <- topic_names
+vizDataFrame <- melt(cbind(data.frame(topic_prop_examples), 
```

Comment:
maybe `pivot_longer` function can be used instead so that we stay within the `tidyverse` universe?